apps/projects/routes.py

```diff
@@ -89,24 +94,22 @@ def post(self):
         if codec_name not in app.config.get('CODEC_SUPPORT'):
             return bad_request("Codec: {} is not supported.".format(codec_name))
 
-        # put file into storage
+        # generate file path
         file_name = create_file_name(ext=file.filename.split('.')[1])
-        mime_type = file.mimetype
-
-        # get path group by year month
-        create_date = datetime.utcnow()
-        folder = f'{create_date.year}/{create_date.month}'
+        utcnow = datetime.utcnow()
+        folder = f'{utcnow.year}/{utcnow.month}'
+        file_path = os.path.join(app.config.get('FS_MEDIA_STORAGE_PATH'), folder, file_name)
```

Comment:
imo this logic should be in `file_system_storage`, it might not make sense for other implementations. not sure about this folder part

Reply:
Yes, first time I put "group folder" into file storage, we only put "file name" into storage.  But when we get this file, we must need a "file path". IMO I see it isn't homologous.
Should it make sense, when we put a 'filename', but when we get a file path?
@petrjasek

Reply:
we discussed this with @ride90 and it could be a bit different, idea is that we store some `storage_id` which should be later used for retrieving the file:
```
doc['storage_id'] = app.fs.put(file_stream, filename='x', content_type='image/jpeg')
binary = app.fs.get(doc['storage_id'])
```
this should work also for other implementation like s3/gridfs.
for file_storage that `storage_id` should be probably `folder/filename` as in this PR

Reply:
we don't use content_type. So we will set it  
```
def put(file_stream, filename='x'):
```
 or for other implementation s3/ griffs. We will set
```
def put(file_stream, filename='x', content_type= None):
```
set content_type default value = None and never use it in file system storage.
@petrjasek

Reply:
yep `content_type` can be ignored by file system backend

Reply:
ok, thanks.